serialization/python/json/serialization/json_parse_node.py

```diff
@@ -0,0 +1,238 @@
+        def func(item):
+            current_parse_node = JsonParseNode(item)
+            if isinstance(item, bool):
+                return current_parse_node.get_boolean_value()
+            elif isinstance(item, str):
+                return current_parse_node.get_string_value()
+            elif isinstance(item, int):
+                return current_parse_node.get_int_value()
+            elif isinstance(item, float):
+                return current_parse_node.get_float_value()
+            elif isinstance(item, UUID):
+                return current_parse_node.get_uuid_value()
+            elif isinstance(item, datetime):
+                return current_parse_node.get_datetime_offset_value()
+            elif isinstance(item, timedelta):
+                return current_parse_node.get_timedelta_value()
+            elif isinstance(item, date):
+                return current_parse_node.get_time_value()
+            elif isinstance(item, time):
+                return current_parse_node.get_time_value()
+            else:
+                raise Exception(
+                    f"Encountered an unknown type during deserialization {type(item)}"
+                )
+
+        return list(map(func, list(self._json_node)))
+
+    def get_collection_of_object_values(self, class_type: Type[U]) -> List[U]:
+        """Gets the collection of type U values from the json node
+        Returns:
+            List[U]: The collection of model object values of the node
+        """
+        def func(item):
+            current_parse_node = JsonParseNode(item)
+            return current_parse_node.get_object_value(class_type)
+
+        return list(map(func, list(self._json_node)))
+
+    def get_collection_of_enum_values(self, type: Any) -> List[K]:
+        """Gets the collection of enum values of the json node
+        Returns:
+            List[K]: The collection of enum values
+        """
+        raw_values = str(self._json_node)
+        if not raw_values:
+            return []
+        return list(map(lambda x: type(x.capitalize()), raw_values.split(",")))
+
+    def get_enum_value(self, type: Any) -> Enum:
+        """Gets the enum value of the node
+        Returns:
+            Enum: The enum value of the node
+        """
+        values = self.get_collection_of_enum_values(type)
+        if values:
+            return values[0]
+        return None
+
+    def get_object_value(self, class_type: Type[U]) -> U:
+        """Gets the model object value of the node
+        Returns:
+            Parsable: The model object value of the node
+        """
+        result = class_type()
+        if self.on_before_assign_field_values:
+            self.on_before_assign_field_values(result)
+        self.assign_field_values(result)
+        if self.on_after_assign_field_values:
+            self.on_after_assign_field_values(result)
+        return result
+
+    def assign_field_values(item: U) -> None:
+        fields = item.get_field_deserializers()
```

Comment:
there should be some looping involved here, and also values that are found and don't have a matching deserializer should go to the additional data

Reply:
Updated in https://github.com/microsoft/kiota/pull/1186/commits/0f55db2b213f00078bb21c22ac603b1fcb6c1ac4